docs/configuration.md

```diff
@@ -0,0 +1,214 @@
+# Configuration variables
+
+Arctic has several tuning knobs under [arctic/_config.py](https://github.com/manahl/arctic/blob/master/arctic/_config.py) which affect the functionality of certain modules.
+
+Most of these variables are initialized via environment variables, which are explained in the rest of this section.
+
+
+<br><br>
+
+## VersionStore
+
+### STRICT_WRITE_HANDLER_MATCH
+
+Controls if Arctic can only match the intended handlers for the data type. If set to true it prevents falling back to pickling if the matching handler (based on the type of data) can't serialize without objects.
+
+```
+export STRICT_WRITE_HANDLER_MATCH=1
+```
+
+
+
+<br><br>
+
+
+## NdArrayStore
+
+### CHECK_CORRUPTION_ON_APPEND
+
+Enables more thorough sanity checks for detecting data corruption when issuing appends. The checks will introduce a 5-7% performance hit. This is disabled by default.
+
+```
+export CHECK_CORRUPTION_ON_APPEND=1
+```
+
+
+
+<br><br>
+
+
+## Serialization
+
+### ARCTIC_AUTO_EXPAND_CHUNK_SIZE
+
+If a row is too large, then auto-expand the data chunk size from the default _CHUNK_SIZE (it is 2MB). It is disabled by default, and the written DataFrame in its serialized Numpy array form shouldn't exceed 2MB.
+
+This setting is effective only when using the incremental serializer.
+
+```
+export ARCTIC_AUTO_EXPAND_CHUNK_SIZE=1
+```
+
+
+### MAX_DOCUMENT_SIZE
+
+This configuration variable is used only when ARCTIC_AUTO_EXPAND_CHUNK_SIZE is set and the user writes a DataFrame which has extremely large number of columns (exceed 2MB serialized).
+
+This value must be less than 16MB which is the maximum document size of MongoDB, taking into account the size of other fields, serialized as a BSON object.
+
+Its default value is configured as follows:
+
+```
+In[8]: pymongo.common.MAX_BSON_SIZE * 0.8
+Out[8]: 13421772.8
+```
+
+
+### FAST_CHECK_DF_SERIALIZABLE
+
+Optional optimisation feature. When set, it applies a fast check for the *can_write()* of the Pandas-specific store implementations. This check takes place to decide the right write handler, among the registered ones, applied on the data provided by the user to be written/appended. It has a significant impact for large DataFrames, which have columns with object dtype. The benefits are even more evident if the number of object columns is proportionally small to the total number of columns.
+
+```
+export FAST_CHECK_DF_SERIALIZABLE=1
+```
+
+
+
+
+
+<br><br>
+
+
+
+## Forward pointers
+
+### ARCTIC_FORWARD_POINTERS_CFG
+
+This feature flag controls the mode of operation for the data segment references model.
```

Comment:
Should we mark this clearly e.g. orange text as experimental?

Reply:
highlighted